mdoc-sbt/src/main/scala/mdoc/MdocPlugin.scala

```diff
@@ -106,21 +112,52 @@ object MdocPlugin extends AutoPlugin {
         mdocInternalVariables.value.foreach { case (key, value) =>
           props.put(key, value)
         }
+        def getJars(mid: ModuleID) = {
+
+          val depRes = dependencyResolution.in(update).value
+          val updc = updateConfiguration.in(update).value
+          val uwconfig = unresolvedWarningConfiguration.in(update).value
+          val modDescr = depRes.wrapDependencyInModule(mid)
+
+          depRes
+            .update(
+              modDescr,
+              updc,
+              uwconfig,
+              streams.value.log
+            )
+            .map(_.allFiles)
+            .fold(uw => throw uw.resolveException, identity)
+        }
+
+        val binaryVersion = scalaBinaryVersion.value
+        val log = streams.value.log
+        val libraries = mdocJSLibraries.value.map(_.data)
+        val workerClasspathOverride = mdocJSWorkerClasspath.value
+
         mdocJSCompileOptions.value.foreach { options =>
-          props.put(
-            s"js-scalac-options",
-            options.options.mkString(" ")
-          )
-          props.put(
-            s"js-classpath",
-            options.classpath.mkString(File.pathSeparator)
-          )
-          options.moduleKind.foreach { moduleKind => props.put(s"js-module-kind", moduleKind) }
+          val sjsVersion = detectScalaJSVersion
+
+          val linkerDependency = binaryVersion match {
+            case "3" => "org.scala-js" % "scalajs-linker_2.13" % sjsVersion
+            case other => "org.scala-js" % s"scalajs-linker_$other" % sjsVersion
+          }
+
+          val mdocJSDependency = binaryVersion match {
+            case "3" => "org.scalameta" % "mdoc-js-worker_3" % BuildInfo.version
+            case other => "org.scalameta" % s"mdoc-js-worker_$other" % BuildInfo.version
+          }
+
+          val workerClasspath = workerClasspathOverride.getOrElse(getJars(mdocJSDependency))
+
+          MdocJSConfiguration(
+            scalacOptions = options.options,
+            compileClasspath = options.classpath,
+            linkerClassPath = getJars(linkerDependency) ++ workerClasspath,
+            moduleKind = options.moduleKind,
+            jsLibraries = libraries
+          ).writeTo(props)
         }
-        props.put(
-          s"js-libraries",
-          mdocJSLibraries.value.map(_.data).mkString(File.pathSeparator)
-        )
```

Comment:
If I were to guess, this setting was extracted from the block above because SBT disallows invoking the task in anonymous lambda, and **not** because it's a valid usecase to not have mdoc.js configuration but still use js-libraries?